src/hooks/useAlgorithmRunnerWithSteps.ts

```diff
@@ -0,0 +1,17 @@
+import { useCallback, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { AlghorithmRunParams, AlgorithmRunnerWithSteps } from "../logic/algo/runnerWithSteps";
+import { ApplicationState } from "../state/appReducer";
+import { SetRunParams } from "../state/runParamsReducer";
+
+export const initialRunParams: AlghorithmRunParams = {
+  branchFactor: 8,
+  treeDepth: 5,
+  numRollout: 5,
+  numIterations: 5,
+};
+
+export function useAlgorithmRunnerWithSteps(): Generator<unknown, unknown, unknown> {
+  const [generator, setGenerator] = useState(AlgorithmRunnerWithSteps.run(initialRunParams));
+  return generator;
+}
```

Comment:
Zostawiłam ten plik useAlgorithmRunnerWithSteps, nie wiem za bardzo jak mogłabym to wpakować do useAlgorithmRunner :(